superset/utils/pandas_postprocessing.py

```diff
@@ -0,0 +1,389 @@
+           on rolling values calculated from `y`, leaving the original column `y`
+           unchanged.
+    :param rolling_type: Type of rolling window. Any numpy function will work.
+    :param rolling_type_options: Optional options to pass to rolling method. Needed
+           for e.g. quantile operation.
+    :param center: Should the label be at the center of the window.
+    :param win_type: Type of window function.
+    :param window: Size of the window.
+    :param min_periods:
+    :return: DataFrame with the rolling columns
+    :raises ChartDataValidationError: If the request in incorrect
+    """
+    rolling_type_options = rolling_type_options or {}
+    df_rolling = df[columns.keys()]
+    kwargs: Dict[str, Union[str, int]] = {}
+    if not window:
+        raise QueryObjectValidationError(_("Undefined window for rolling operation"))
+
+    kwargs["window"] = window
+    if min_periods is not None:
+        kwargs["min_periods"] = min_periods
+    if center is not None:
+        kwargs["center"] = center
+    if win_type is not None:
+        kwargs["win_type"] = win_type
+
+    df_rolling = df_rolling.rolling(**kwargs)
+    if rolling_type not in WHITELIST_ROLLING_FUNCTIONS or not hasattr(
+        df_rolling, rolling_type
+    ):
+        raise QueryObjectValidationError(
+            _("Invalid rolling_type: %(type)s", type=rolling_type)
+        )
+    try:
+        df_rolling = getattr(df_rolling, rolling_type)(**rolling_type_options)
+    except TypeError:
+        raise QueryObjectValidationError(
+            _(
+                "Invalid options for %(rolling_type)s: %(options)s",
+                rolling_type=rolling_type,
+                options=rolling_type_options,
+            )
+        )
+    df = _append_columns(df, df_rolling, columns)
+    if min_periods:
+        df = df[min_periods:]
+    return df
+
+
+@validate_column_args("columns", "rename")
+def select(
+    df: DataFrame, columns: List[str], rename: Optional[Dict[str, str]] = None
+) -> DataFrame:
+    """
+    Only select a subset of columns in the original dataset. Can be useful for
+    removing unnecessary intermediate results, renaming and reordering columns.
+
+    :param df: DataFrame on which the rolling period will be based.
+    :param columns: Columns which to select from the DataFrame, in the desired order.
+                    If columns are renamed, the new column name should be referenced
+                    here.
+    :param rename: columns which to rename, mapping source column to target column.
+                   For instance, `{'y': 'y2'}` will rename the column `y` to
+                   `y2`.
+    :return: Subset of columns in original DataFrame
+    :raises ChartDataValidationError: If the request in incorrect
+    """
+    df_select = df[columns]
+    if rename is not None:
+        df_select = df_select.rename(columns=rename)
+    return df_select
```

Comment:
@villebro can we change this to rename first, then select? More of the time, rename maps existing columns to a more readable name, and it'd be more convenient to select from the more desirable output column name.